example/lib/peripheral_details/peripheral_details_screen.dart

```diff
@@ -9,48 +9,72 @@ class PeripheralDetailsScreen extends StatelessWidget {
   Widget build(BuildContext context) {
     return BlocBuilder<PeripheralDetailsBloc, PeripheralDetailsState>(
       builder: (context, state) {
-        if (state.peripheral.category == BlePeripheralCategory.sensorTag) {
-          return DefaultTabController(
-            length: 3,
-            child: Scaffold(
-              appBar: AppBar(
-                title: Text(state.peripheral.name),
-                bottom: TabBar(tabs: [
-                  Tab(
-                    icon: Icon(Icons.table_chart),
-                    text: 'Details',
-                  ),
-                  Tab(
-                    icon: Icon(Icons.format_list_numbered),
-                    text: 'Auto test',
-                  ),
-                  Tab(
-                    icon: Icon(Icons.settings),
-                    text: 'Manual test',
-                  ),
-                ]),
-              ),
-              body: TabBarView(
-                children: <Widget>[
-                  _buildDetailsView(),
-                  Text('Auto test'),
-                  Text('Manual test'),
-                ],
-              ),
-            ),
-          );
+        if (state is PeripheralAvailable) {
+          return _buildForPeripheralAvailable(state);
+        } else if (state is PeripheralUnavailable) {
+          return _buildForPeripheralUnavailable(state);
         } else {
-          return Scaffold(
-            appBar: AppBar(
-              title: Text(state.peripheral.name),
-            ),
-            body: _buildDetailsView(),
-          );
+          return null;
         }
       },
     );
   }
 
+  Widget _buildForPeripheralAvailable(PeripheralAvailable state) {
+    if (state.peripheralInfo.peripheralLayout ==
```

Comment:
If there's an if with two long branches, you'd be better off having those branches in separate functions. Makes for an easier read.